src/pages/index.js

```diff
@@ -16,79 +22,241 @@ const cardListEl = document.querySelector(".cards__list");
-}
+//api instantiate
 
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "75623fcf-ddee-44c9-84c0-daef735bad5c",
+    "Content-Type": "application/json",
   },
-  ".cards__list"
-);
+});
+
+//api on page loading
+
+api.getInitialCards().then((initialCards) => {
+  section = new Section(
+    {
+      items: initialCards,
+      renderer: renderCard,
+    },
+    ".cards__list"
+  );
+  section.renderItems();
+});
+
+api
+  .getUserInfo()
+  .then((userInfoData) => {
+    userInfo.setUserInfo(userInfoData);
+    userInfo.setUserProfileImage(userInfoData.avatar);
+  })
 
-section.renderItems();
+  .catch((err) => {
+    console.error(err);
+  });
+
+//instantiating other classes
 
 const popupWithImage = new PopupWithImage("#card-image");
-popupWithImage.setEventListeners();
 
 const newCardModal = new PopupWithForm(
   "#add-card-modal",
   handleAddCardFormSubmit
 );
-newCardModal.setEventListeners();
 
 const editCardModal = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
+
+const deleteCardModal = new DeleteConfirmation(
+  "#delete-card-confirmation",
+  deleteCardForm,
+  handleDeleteCard
+);
+
+const profilePictureModal = new PopupWithForm(
+  "#change-profile-picture-modal",
+  handleChangeProfilePictureSubmit
+);
+
+// set event listeners
+
+popupWithImage.setEventListeners();
+
+newCardModal.setEventListeners();
+
 editCardModal.setEventListeners();
 
-const userInfo = new UserInfo({
-  titleSelector: ".profile__title",
-  descriptionSelector: ".profile__description",
+deleteCardModal.setEventListeners();
+
+profilePictureModal.setEventListeners();
+
+//button Event Listeners
+
+profileEditButton.addEventListener("click", () => {
+  const userData = userInfo.getUserInfo();
+  profileTitleInput.value = userData.title;
+  profileDescriptionInput.value = userData.description;
+  editCardModal.open();
 });
 
-function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data);
-  editCardModal.close();
+profileImageEditButton.addEventListener("click", () => {
+  profilePictureModal.open();
+});
+
+addNewCardButton.addEventListener("click", () => newCardModal.open());
+
+//functions
+
+function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    cardSelector,
+    showPreviewImage,
+    handleLikeCard,
+    handleDeleteCardPopup
+  );
+  section.addItem(card.getView());
+}
+
+function handleAddCardFormSubmit({ title, url }) {
+  addCardModalSubmitButton.innerHTML = "Saving...";
+  api
+    .postCard({ name: title, link: url })
+    .then((cardData) => {
+      renderCard(cardData, cardListEl);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardFormElement.reset();
+      addCardModalSubmitButton.classList.add("modal__button_disabled");
+      newCardModal.close();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well


**Please, correct it everywhere**